source/opt/mem_pass.cpp

```diff
@@ -195,8 +195,8 @@ void MemPass::DCEInst(ir::Instruction* inst,
       continue;
     }
     // Remember operands
-    std::vector<uint32_t> ids;
-    di->ForEachInId([&ids](uint32_t* iid) { ids.push_back(*iid); });
+    std::set<uint32_t> ids;
+    di->ForEachInId([&ids](uint32_t* iid) { ids.insert(*iid); });
```

Comment:
Consider
const uint32_t* iid

Reply:
Yes that would be better, but I don't think it is worth waiting for the build bot again.